src/async_impl/response.rs

```diff
@@ -45,17 +51,29 @@ impl Response {
         &mut self.headers
     }
 
-    /// Get a mutable reference to the `Body` of this `Response`.
+    /// Get a readable response body.
+    /// 
+    /// The response will be decoded.
     #[inline]
-    pub fn body_mut(&mut self) -> &mut Body {
+    pub fn body_mut(&mut self) -> &mut Decoder {
         &mut self.body
     }
 
+    /// Get a readable response body.
+    /// 
+    /// This function will replace the body on the response with an empty one.
+    #[inline]
+    pub fn body(&mut self) -> Decoder {
```

Comment:
I actually would have `body() -> &Decoder`, to match the `http` crate...

Taking a `Decoder` can be done with `mem::swap` and `body_mut()`, right?